packages/vx-xychart/src/hooks/useDataRegistry.ts

```diff
@@ -0,0 +1,13 @@
+import { AxisScaleOutput } from '@vx/axis';
+import { useMemo } from 'react';
+import { ScaleConfig } from '@vx/scale';
+import DataRegistry from '../classes/DataRegistry';
+
+/** Hook that returns a constant instance of a DataRegistry.  */
+export default function useDataRegistry<
+  XScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  YScaleConfig extends ScaleConfig<AxisScaleOutput>,
+  Datum = unknown
```

Comment:
(optional) Not sure if `Datum` should be `object` instead of `unknown`? But that won't allow primitive types to be `Datum`.

Reply:
good question, could `extend object = object`. I also thought about primitives, I'm not sure how that could be useful in an xychart context unless you had a constant `x` or `y` (in which case the accessor could return that constant) ... which might be valid. maybe should go for `object` for now and can loosen the restriction in the future.